samples/nrf9160/modem_shell/src/location/location_service_nrf_cloud.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <cJSON.h>
+#include <net/nrf_cloud.h>
+#include <net/nrf_cloud_agps.h>
+#include <net/nrf_cloud_pgps.h>
+#include <net/nrf_cloud_location.h>
+#include <modem/location.h>
+
+#include "mosh_print.h"
+#include "mosh_defines.h"
+
+static struct location_data nrf_cloud_location;
+static K_SEM_DEFINE(location_ready, 0, 1);
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+void location_service_agps_ext_handle(const struct nrf_modem_gnss_agps_data_frame *agps_req)
+{
+	int err;
+
+	err = nrf_cloud_agps_request(agps_req);
+	if (err) {
+		mosh_error("nRF Cloud A-GPS request failed, error: %d", err);
+		return;
+	}
+
+	mosh_print("A-GPS data requested");
+}
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+
+#define NRF_CLOUD_JSON_APPID_KEY		"appId"
+#define NRF_CLOUD_JSON_APPID_VAL_PGPS		"PGPS"
+#define NRF_CLOUD_JSON_MSG_TYPE_KEY		"messageType"
+#define NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA	"DATA"
+#define NRF_CLOUD_JSON_DATA_KEY			"data"
+#define NRF_CLOUD_JSON_PGPS_PRED_COUNT		"predictionCount"
+#define NRF_CLOUD_JSON_PGPS_INT_MIN		"predictionIntervalMinutes"
+#define NRF_CLOUD_JSON_PGPS_GPS_DAY		"startGpsDay"
+#define NRF_CLOUD_JSON_PGPS_GPS_TIME		"startGpsTimeOfDaySeconds"
```

Comment:
We should have only one header for these json key definitions. I.e. in nrf cloud side. Otherwise we need to update all these if something changes.

Reply:
There is internal header nrf_cloud_codec.h which I'm currently not including in the sample as I think it's questionable.

@plskeggs How should we handle these? I'm seeing that nrf_cloud_mqtt_multi_service sample includes this header to get access to some of these defines:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c#L16

I have also copied encoding of the P-GPS request into this file from pgps_request() function. I guess this is somewhat fine.

Reply:
This is maybe a stupid question but why not just include nrf_cloud_codec.h?

Reply:
`nrf_cloud_codec.h` is not within `sdk-nrf/include` but it's internal to nrf_cloud library at `sdk-nrf/subsys/net/lib/nrf_cloud/include`. I'd expect that those shouldn't be used although we do it in nrf.

Reply:
@jayteemo are you ok with moving those JSON definitions to a new header in the sdk_nrf public includes folder?

Reply:
yes, i thought we even had a ticket to do that?

Reply:
@trantanen I've created a ticket NCSDK-19170 to move the JSON definitions to the public folder.  I will also create one to move the P-GPS request encoding to the codec file.

Go ahead and include the private header for now, and we'll clean it up once we take care of the ticket.